File: HotDesk_SPICE/TMS_HotDesk/Backup/Library/Library.Database/BLL/MenuNav.vb
Namespace BLL
    Public Class MenuNav
        Inherits Library.Root.Other.BusinessLogicBase

        Public Shared Function List(ByVal Table As String, ByVal SearchField As String, ByVal SearchValue As String, ByVal SortField As String, ByVal Direction As Integer, _
                                   ByVal Page As Integer, ByVal Deleted As Integer) As ListCollection
            Using _dal As New DAL.MenuNav
                'Validation the parameter value
                If Direction <> 1 Then
                    Direction = 0
                End If

                List = _dal.List(Table, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted)
            End Using
        End Function

                  Public Shared Function GetIDList() As DataTable
                           Using _dal As New DAL.MenuNav
                                    GetIDList = _dal.GetIDList()
                           End Using
                  End Function

                  Public Shared Function GetPalletData() As DataTable
                           Using _dal As New DAL.MenuNav
                                    GetPalletData = _dal.GetPalletData()
                           End Using
                  End Function

        Public Shared Function GetData(ByVal ID As String) As DataTable
            Using _dal As New DAL.MenuNav
                GetData = _dal.GetData(ID)
            End Using
        End Function


        Public Shared Function Maint(ByVal ID As String, ByVal mods As String, ByVal RecType As String) As String
            Using _Dal As New DAL.MenuNav
                Dim str As String = System.Web.HttpContext.Current.Session("gstrUserID").ToString
                Dim cc As String = System.Web.HttpContext.Current.Session("gstrUserCompCode").ToString
                Maint = _Dal.Maint(ID, mods, RecType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString, cc)

                If Maint = "1" Then
                    _Dal.Commit()
                Else
                    _Dal.RollBack()
                End If
            End Using
        End Function

                  
         End Class

         
End Namespace

File: HotDesk_SPICE/Library/Library.Database/BLL/MenuNav.cs
using System.Data;
using Library.Database;

namespace BLL
{
    public class MenuNav : Library.Root.Other.BusinessLogicBase
    {
        public static ListCollection List(string table, string searchField, string searchValue, string sortField, int direction, int page, int deleted)
        {
            using (var _dal = new DAL.MenuNav())
            {
                if (direction != 1)
                {
                    direction = 0;
                }

                return _dal.List(table, searchField, searchValue, sortField, direction, FromRowNo(page), ToRowNo(page), deleted);
            }
        }

        public static DataTable GetIDList()
        {
            using (var _dal = new DAL.MenuNav())
            {
                return _dal.GetIDList();
            }
        }

        public static DataTable GetPalletData()
        {
            using (var _dal = new DAL.MenuNav())
            {
                return _dal.GetPalletData();
            }
        }

        public static DataTable GetData(string id)
        {
            using (var _dal = new DAL.MenuNav())
            {
                return _dal.GetData(id);
            }
        }

        public static string Maint(string id, string mods, string recType)
        {
            using (var _dal = new DAL.MenuNav())
            {
                string str = System.Web.HttpContext.Current.Session["gstrUserID"].ToString();
                string cc = System.Web.HttpContext.Current.Session["gstrUserCompCode"].ToString();
                string result = _dal.Maint(id, mods, recType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString(), cc);

                if (result == "1")
                {
                    _dal.Commit();
                }
                else
                {
                    _dal.Rollback();
                }

                return result;
            }
        }
    }
}